### DISCUSSION TITLE
How do I get the cyclomatic complexity of a file before change?

### DISCUSSION THREAD

**Discussion by ks3057** [ishepard/pydriller#167] [Q&A] [answered]
Hi! I can see the Modification object returns the cyclomatic complexity of a file changed in a commit. Is there a way to check what it was previously? Basically, I'd like to get a diff of the cyclomatic complexity for comparison.

**Comment by Dlitosh** [comment, marked as answer]
@ks3057 perhaps you can use this method https://github.com/ishepard/pydriller/blob/2ad53b0544e765187d4a23fdb4096d26bd663afb/pydriller/domain/commit.py#L436 with a flag include_before=True?

**Comment by ishepard** [comment]
Hi @ks3057, sorry I haven't replied sooner, I missed this discussion.

Yes you are right, the complexity at file level is only calculated after the change, while the complexity at method level can be calculated before and after. I think we can add the complexity at file level before. What do you think, can you work on a PR?
